### PR TITLE
feat: upgrade golang-jwt to v5

### PR DESCRIPTION
**Pull request by ilaleksin**
## Description

This PR addresses https://github.com/openfga/openfga/issues/1396
It upgrades **golang-jwt to v5** and **MicahParks/keyfunc to v2.0.3** 

- refactored JWT claims to align with jwt v5 interface.
- added a separate **WithIssuedAt** validation because the iat claim check was [removed from jwt.Parse() function](https://github.com/golang-jwt/jwt/blob/main/MIGRATION_GUIDE.md#parsing-and-validation-options)
- added one unit test to verify a case when iss alias is set but the JWT iss claim has the original issuer value
- keyfunc is upgraded to v2 because it depends on golang-jwt v5 while keyfunc v1 depends on golang-jwt v4

## References

https://github.com/golang-jwt/jwt/blob/main/MIGRATION_GUIDE.md#parsing-and-validation-options
https://github.com/MicahParks/keyfunc/releases/tag/v2.0.0

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: internal/authn/oidc/oidc.go [hunk 1]
```diff
@@ -83,26 +83,29 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
...
 		return nil, errInvalidClaims
 	}
 
-	validIssuers := []string{
-		oidc.MainIssuer,
+	if err := jwt.NewValidator(jwt.WithIssuedAt()).Validate(claims); err != nil {
+		return nil, errInvalidToken
 	}
+
+	validIssuers := []string{oidc.MainIssuer}
```
Comment (adriantam) [inline internal/authn/oidc/oidc.go:95]: nitpick: this line can be combined with the next line.

Comment (ilaleksin) [reply internal/authn/oidc/oidc.go]: changed to `validIssuers := append(oidc.IssuerAliases, oidc.MainIssuer)`

is it ok?

File: go.mod [hunk 2]
```diff
@@ -4,13 +4,13 @@ go 1.21.9
 
 require (
 	github.com/Masterminds/squirrel v1.5.4
-	github.com/MicahParks/keyfunc v1.9.0
+	github.com/MicahParks/keyfunc/v2 v2.0.3
```
Comment (miparnisari) [inline go.mod:7]: Please upgrade to v3: https://github.com/MicahParks/keyfunc/tree/v3.3.2

File: internal/authn/oidc/oidc.go [hunk 3]
```diff
@@ -83,26 +83,28 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
...
 	}
 
-	validIssuers := []string{
-		oidc.MainIssuer,
+	if err := jwt.NewValidator(jwt.WithIssuedAt()).Validate(claims); err != nil {
+		return nil, errInvalidToken
 	}
-	validIssuers = append(validIssuers, oidc.IssuerAliases...)
+
+	validIssuers := append(oidc.IssuerAliases, oidc.MainIssuer)
```
Comment (miparnisari) [inline internal/authn/oidc/oidc.go:95]: Why did you change this from

```go
validIssuers := []string{
	oidc.MainIssuer,
}
validIssuers = append(validIssuers, oidc.IssuerAliases...)
``` 

?

File: internal/authn/oidc/oidc.go [hunk 4]
```diff
@@ -83,26 +83,28 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
...
 	claims, ok := token.Claims.(jwt.MapClaims)
 	if !ok {
 		return nil, errInvalidClaims
 	}
 
-	validIssuers := []string{
-		oidc.MainIssuer,
+	if err := jwt.NewValidator(jwt.WithIssuedAt()).Validate(claims); err != nil {
+		return nil, errInvalidToken
 	}
```
Comment (miparnisari) [inline internal/authn/oidc/oidc.go:93]: You can get rid of this block and change line 74 to

```go
jwtParser := jwt.NewParser(
		jwt.WithValidMethods([]string{jwt.SigningMethodRS256.Alg()}),
		jwt.WithIssuedAt(),
	)
```

Also, please add `jwt.WithExpirationRequired(),` and add a test for that.

File: internal/authn/oidc/oidc.go [hunk 5]
```diff
@@ -83,26 +83,28 @@ func (oidc *RemoteOidcAuthenticator) Authenticate(requestContext context.Context
...
+		return err == nil
 	})
 
 	if !ok {
 		return nil, errInvalidIssuer
 	}
 
-	if ok := claims.VerifyAudience(oidc.Audience, true); !ok {
+	if err := jwt.NewValidator(jwt.WithAudience(oidc.Audience)).Validate(claims); err != nil {
 		return nil, errInvalidAudience
```
Comment (miparnisari) [inline internal/authn/oidc/oidc.go:108]: Thought (not for this PR): i think we shot ourselves in the foot by having all these custom error codes. The code would be much simpler and therefore more secure if we performed all validations in one go:

```go
jwtParser := jwt.NewParser(
		jwt.WithValidMethods([]string{jwt.SigningMethodRS256.Alg()}),
		jwt.WithIssuedAt(),
		jwt.WithExpirationRequired(),
		jwt.WithAudience(config.Audience),
	)
if err != nil  || !token.Valid{
	return nil, errInvalidToken
}
```

:(

**Review by miparnisari — CHANGES_REQUESTED** [review]

**Review by miparnisari — APPROVED** [review]

**Review by sergiught — APPROVED** [review]
